### PR TITLE
mono: initial recipe

### PR DESCRIPTION
**Pull request by hundeboll**
add the recipe to bootstrap mono and a very simple hello-world recipe to make use of the just added mono.

### CODE REVIEW

File: recipes/helloworld/mono-helloworld_git.oe.sig [hunk 1]
```diff
@@ -0,0 +1 @@
+137c01b31301a730f488ecd57d02c96615fbc072  e2286f3e291c5d88f5fe8ef679c2f34bf34f7f7d.zip
```
Comment (kimrhh) [inline recipes/helloworld/mono-helloworld_git.oe.sig]: [minor] hmm, why not just pull it directly from git if there are no releases anyway?

**Comment by kimrhh** [comment]
:+1: very nice work, especially adding a working example program for the runtime :)

File: recipes/helloworld/mono-helloworld_git.oe.sig (hunk 1 above)
Comment (hundeboll) [inline recipes/helloworld/mono-helloworld_git.oe.sig]: Ok, I changed it to git...

File: recipes/mono/mono.inc [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    libpthread \
+    libc \
+    libgcc-s \
+    librt \
+    libstdc++ \
+    libmono-2.0 \
+    libmonosgen-2.0 \
+    libmonosupportw \
+    libmonoposixhelper \
+    ikvm-native \
```
Comment (esben) [inline recipes/mono/mono.inc]: ikvm-native or libikvm-native?

File: recipes/mono/mono.inc [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    monosupportw \
+    ikvm-native \
+    monoposixhelper \
+"
+AUTO_PACKAGE_LIBS:>machine = " monoboehm-2.0"
+AUTO_PACKAGE_LIBS_DEPENDS = "libc libdl libgcc libm libpthread librt libstdc++"
+AUTO_PACKAGE_LIBS_RDEPENDS = "libc libdl libgcc libm libpthread librt libstdc++"
+
+# libmono-2.0
+LIBRARY_VERSION_${PN}-libmono-2.0 = "1.0.0"
```
Comment (esben) [inline recipes/mono/mono.inc]: Is this the right LIBRARY_VERSION?  And if so, why is it necessary to add libmono-2.0-1.0.0 to allow-missing-soname?

File: recipes/mono/mono.inc (hunk 2 above)
Comment (hundeboll) [inline recipes/mono/mono.inc]: the latter.

File: recipes/mono/mono.inc (hunk 3 above)
Comment (hundeboll) [inline recipes/mono/mono.inc]: You are right. I thought it was more or less mandatory to specify library versions, but removing those made the qa flags unnecessary.

**Pull request merged**